src/Features/Core/Portable/QuickInfo/CommonSemanticQuickInfoProvider.cs

```diff
@@ -308,6 +320,42 @@ void AddSection(string kind, ImmutableArray<TaggedText> taggedParts)
             return default;
         }
 
+        private ImmutableArray<TaggedText> GetReturnsDocumentationContent(
+            IEnumerable<ISymbol> symbols,
+            IDictionary<SymbolDescriptionGroups, ImmutableArray<TaggedText>> sections,
+            SemanticModel semanticModel,
+            SyntaxToken token,
+            IDocumentationCommentFormattingService formatter,
+            ISyntaxFactsService syntaxFactsService,
+            CancellationToken cancellationToken)
+        {
+            if (sections.TryGetValue(SymbolDescriptionGroups.ReturnsDocumentation, out var parts))
+            {
+                var documentationBuilder = new List<TaggedText>();
```

Comment:
ImmutableArray.Builder instead of List?

Reply:
Or just return the array directly? is AddRange doing something fancy here?

Reply:
Returning the array directly cleans things up nicely.